### PR TITLE
Merge patch into develop

### PR DESCRIPTION
**Pull request by janinefreeman**
Hi team- please review all things that you worked on in patch and **comment** on the pull request if something that Git has decided should be merged into develop is incorrect. (Do this by looking at "files changed"). Please comment on the PR once you have reviewed, so that I know when we're ready to merge. I will handle the final merging of patch into develop. After that, every time you update patch, please then merge your changes into develop.

### CODE REVIEW

File: nlopt/config - win.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/* config.h.  Generated from config.h.in by configure.  */
```
Comment (dguittet) [inline nlopt/config - win.h:1]: This should be removed from patch & develop as it's developer-specific and should be generated individually by the developer by `configure` during the build process.

File: build_sdk/apiwrappers/csharp/App.config [hunk 2]
```diff
@@ -1,6 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8" ?>
-<configuration>
```
Comment (sjanzou) [inline build_sdk/apiwrappers/csharp/App.config:2]: Why did these reappear? All code wrappers are now done from the SAM code generation menu.

File: build_sdk/apiwrappers/python/openei_utilityrate_billcalculator.py [hunk 3]
```diff
@@ -1,344 +0,0 @@
-#######################################################################
-## SDK Example: PVWatts + Utility Rate
```
Comment (sjanzou) [inline build_sdk/apiwrappers/python/openei_utilityrate_billcalculator.py:2]: Should we keep this somewhere - SAM-documentation or SAM/samples or website LK Cookbook?

File: build_sdk/examples/6parsolve.lk [hunk 4]
```diff
@@ -1,21 +0,0 @@
-var('Vmp', 30);
-var('Imp', 6);
```
Comment (sjanzou) [inline build_sdk/examples/6parsolve.lk:2]: Again - maybe an LK Cookbook?

File: build_sdk/examples/import_SAM_inputs.lk [hunk 5]
```diff
@@ -1,70 +0,0 @@
...
-			 the variables and their values into the SSCDev workspace
-			 
-Input      - path2input: the absolute pathname of the input file, eg:
-			 path2input = 'U:/CraigFiles/sscdbg_swh.txt';
-			 
-Outputs    - none, loads variables into workspace
-
-Notes      - Does not handle 'table' variables, e.g. PV model shading inputs.
-
-Modified   - Nick DiOrio: 6-28-2013
```
Comment (sjanzou) [inline build_sdk/examples/import_SAM_inputs.lk:15]: Maybe keep this for reference?

File: shared/lib_battery.cpp [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
-WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT 
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED.IN NO EVENT SHALL THE COPYRIGHT HOLDER, CONTRIBUTORS, UNITED STATES GOVERNMENT OR UNITED STATES
+DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+OR CONSEQUENTIAL DAMAGES(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
 OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
```
Comment (sjanzou) [inline shared/lib_battery.cpp:22]: These have ssc tests with fixes so should be good!

File: shared/lib_battery_dispatch.cpp [hunk 7]
```diff
@@ -61,7 +61,7 @@ dispatch_t::dispatch_t(battery_t * Battery, double dt_hour, double SOC_min, doub
 
 void dispatch_t::init(battery_t * Battery, double dt_hour, int current_choice, double t_min, int mode)
 {
-	_dt_hour = dt_hour; 
+	_dt_hour = dt_hour;
 	_current_choice = current_choice;
 	_t_min = t_min;
```
Comment (sjanzou) [inline shared/lib_battery_dispatch.cpp:66]: Just checking that all battery fixes have been merged into patch and not develop - correct?

File: ssc/cmod_pvwattsv7.cpp [hunk 8]
```diff
@@ -714,7 +714,9 @@ class cm_pvwattsv7 : public compute_module
 #define NSTATUS_UPDATES 50  // set this to the number of times a progress update should be issued for the simulation
 				if (hour % (8760 / NSTATUS_UPDATES) == 0)
 				{
-					float percent = 100.0f * ((float)idx_life + 1) / ((float)nlifetime / 3.0f); //3 is the number of technologies we're assuming for this output (pvwatts + fuel cell + battery)
+					float percent = 100.0f * ((float)idx_life + 1) / ((float)nlifetime ); //3 is the number of technologies we're assuming for this output (pvwatts + fuel cell + battery)
```
Comment (sjanzou) [inline ssc/cmod_pvwattsv7.cpp:717]: This is the change that addressed the progress bar > 300%

File: build_sdk/apiwrappers/python/openei_utilityrate_billcalculator.py (hunk 3 above)
Comment (janinefreeman) [reply build_sdk/apiwrappers/python/openei_utilityrate_billcalculator.py:2]: @cpaulgilman can you please evaluate if any of these files are worth moving to a different location as reference? thanks for the suggestion @sjanzou :)

**Comment by dguittet** [comment]
Why are there two nlopt/config-windows.h and nlopt/config - win/h?

**Comment by dguittet** [comment]
> Nice that 186 files removed from build_sdk. There are a couple of lk scripts noted that could be preserved in the lk cookbook or SAM-documentation.

Sorry @sjanzou ! I was too quick to judge and incorrectly assumed that folder was newly added files. My mistake, and thanks for correcting that.

File: build_sdk/apiwrappers/python/openei_utilityrate_billcalculator.py (hunk 3 above)
Comment (dguittet) [reply build_sdk/apiwrappers/python/openei_utilityrate_billcalculator.py:2]: Sorry I don't think they should be deleted. I thought build_sdk was a build folder (clearly I was wrong).

**Comment by sjanzou** [comment]
> Why are there two nlopt/config-windows.h and nlopt/config - win.h?

@dguittet
I did not see the nlopt/config-win.h file in the repo - where is that one? 
config.h is generated automatically on Linux and macOS and the fails on windows and has to be replaced with the config-windows.h. The issue was that the config.h used to be checked in and would break Windows build and the config-windows.h is necessary since config.h not generated on Windows...

**Comment by dguittet** [comment]
Nevermind, it looks like it was resolved already. There was a duplicate earlier that has been deleted. Thanks.

**Pull request merged**